src/bluesky/utils/__init__.py

```diff
@@ -93,6 +95,24 @@ def __repr__(self):
 #: Scalar or iterable of values, one to be applied to each point in a scan
 ScalarOrIterableFloat = Union[float, TypingIterable[float]]
 
+# Single function to be used as an event listener
+Subscriber = Callable[[str, P], Any]
+
+OneOrMany = Union[P, list[P]]
+
+
+# Mapping from event type to listener or list of listeners
+class SubscriberMap(TypedDict, total=False):
+    all: OneOrMany[Subscriber[DocumentType]]
+    start: OneOrMany[Subscriber[RunStart]]
+    stop: OneOrMany[Subscriber[RunStop]]
+    event: OneOrMany[Subscriber[Event]]
+    descriptor: OneOrMany[Subscriber[EventDescriptor]]
```

Comment:
This looks correct for what the docstring in the RunEngine says, do the subscribers not support the other doc types (Resource[Stream], Datum[Stream]) without going through `all`?